<!DOCTYPE html>
<html lang="en-CA"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.33">

  <meta name="author" content="David Weber, Nat DeFries">
  <title>MICOM EpiData Workshop 2025 ‚Äì day1-morning</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-c10c794533db35eace88283ed89861b0.css">
  <link rel="stylesheet" href="tachyons-minimal.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">


<section class="slide level2">

<div class="flex">
<div class="w-20">

</div>
<section id="section" class="slide level2 w-80" data-background-image="gfx/cover-art-1.svg" data-background-position="bottom">
<h2>Introduction to Panel Data in Epidemiology</h2>
<h3 id="section-1">MICOM Tooling Workshop 2025</h3>
<p><br></p>
<h4 id="section-2">David Weber, Nat DeFries</h4>
<p><span class="fstyle">Adapted from slides by Alice Cima, Rachel Lobay, Daniel McDonald, Ryan Tibshirani, with huge thanks to Logan Brooks, Xueda Shen, and Dmitry Shemetov</span></p>
<p>12 August 2025</p>
</section>
</div>
</section>
<section>
<section id="workshop-overview-and-system-setup" class="title-slide slide level1 center" data-number="1">
<h1><span class="header-section-number">1</span> Workshop Overview and System Setup</h1>

</section>
<section id="what-we-will-cover" class="slide level2">
<h2>What we will cover</h2>
<ul>
<li>Characteristics of panel data in epidemiology</li>
<li>Tools for processing and plotting panel data</li>
<li>Statistical background on nowcasting and forecasting</li>
<li>Tools for building nowcasting and forecasting models</li>
</ul>
</section>

<section id="system-setup-required-software" class="slide level2">
<h2>System setup ‚Äì Required software</h2>
<p><br></p>
<h3 id="we-assume-you-already-have">We assume you already have</h3>
<p><br></p>
<ol type="1">
<li><code>R</code></li>
</ol>
<p><br></p>
<ol start="2" type="1">
<li>An IDE. We‚Äôll use RStudio, but you can use VSCode or Emacs or Whatnot</li>
</ol>
</section>
<section id="system-setup-downloading-the-materials" class="slide level2">
<h2>System setup ‚Äì Downloading the materials</h2>
<!--
TODO
-->
<h3 id="easy-way">Easy way:</h3>
<ol type="1">
<li>Click the <span class="fourth-colour">Big Green Button</span> that says <kbd>&lt; &gt; Code ‚ñæ</kbd></li>
<li>Choose <span class="primary">Download Zip</span></li>
<li>Open the Zip directory and then Open <code>micom-tooling-workshop-2025.Rproj</code></li>
</ol>
<h3 id="more-expert-local-git-user">More expert (local <code>git</code> user):</h3>
<ol type="1">
<li>Click the <span class="fourth-colour">Big Green Button</span> that says <kbd>&lt; &gt; Code ‚ñæ</kbd></li>
<li>Copy the URL.</li>
<li>Open RStudio, select File &gt; New Project &gt; Version Control. Paste there and proceed.</li>
</ol>
<h3 id="even-more-expert-via-github-remote">Even more expert (via <code>github</code> remote):</h3>
<ol type="1">
<li>Click the Grey Button that says <kbd>‚ëÇ Fork ‚ñæ</kbd></li>
<li>Proceed along the same lines as above.</li>
</ol>
</section>
<section id="system-setup-installing-required-packages" class="slide level2">
<h2>System setup ‚Äì Installing required packages</h2>
<!--
TODO
-->
<p>We will use a <span class="tertiary">lot</span> of packages.</p>
<h3 id="weve-tried-to-make-it-so-you-can-get-them-all-at-once-with-the-right-versions">We‚Äôve tried to make it so you can get them all at once (with the right versions)</h3>
<p>ü§û We hope this works‚Ä¶ ü§û <span class="fourth-colour">Note that you can ‚ÄúCopy to Clipboard‚Äù</span></p>
<p><br></p>
<p>In RStudio:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a></a><span class="fu">install.packages</span>(<span class="st">"renv"</span>)</span>
<span id="cb1-2"><a></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><br></p>
<p>Hopefully, you see: ::: {.cell layout-align=‚Äúcenter‚Äù} ::: {.cell-output .cell-output-stderr}</p>
<pre><code>‚úî You should be good to go!</code></pre>
<p>::: :::</p>
<p>Ask for help if you see something like: ::: {.cell layout-align=‚Äúcenter‚Äù} ::: {.cell-output .cell-output-error}</p>
<pre><code>Error in `verify_setup()`:
! The following packages do not have the correct version:
‚Ñπ Installed: epipredict 0.2.0.
‚Ñπ Required: epipredict == 0.1.5.</code></pre>
<p>::: :::</p>
</section></section>
<section>
<section id="panel-data" class="title-slide slide level1 center" data-number="2">
<h1><span class="header-section-number">2</span> Panel Data</h1>

</section>
<section id="panel-data-1" class="slide level2">
<h2>Panel data</h2>
<ul>
<li><p><a href="https://en.wikipedia.org/wiki/Panel_data"><strong>Panel data</strong></a> or longitudinal data, contain cross-sectional measurements of subjects over time.</p></li>
<li><p>Since we‚Äôre working with aggregated data, the subjects are geographic units (e.g.&nbsp;counties, states).</p></li>
</ul>
<!-- and not individuals.-->
<ul>
<li><p>In table form, panel data is a time index + one or more locations/keys.</p></li>
<li><p><strong>Ex</strong>: The % of outpatient doctor visits that are COVID-related in CA from June 2020 to Dec.&nbsp;2021 (<a href="https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/doctor-visits.html">docs</a>): ::: {.cell layout-align=‚Äúcenter‚Äù} ::: {.cell-output .cell-output-stdout}</p></li>
</ul>
<pre><code># A tibble: 549 √ó 3
   time_value geo_value percent_cli
   &lt;date&gt;     &lt;chr&gt;           &lt;dbl&gt;
 1 2020-06-01 ca               2.75
 2 2020-06-02 ca               2.57
 3 2020-06-03 ca               2.48
 4 2020-06-04 ca               2.41
 5 2020-06-05 ca               2.57
 6 2020-06-06 ca               2.63
 7 2020-06-07 ca               2.73
 8 2020-06-08 ca               3.04
 9 2020-06-09 ca               2.97
10 2020-06-10 ca               2.99
# ‚Ñπ 539 more rows</code></pre>
<p>::: :::</p>
</section>
<section id="examples-of-panel-data---covid-19-cases" class="slide level2">
<h2>Examples of panel data - COVID-19 cases</h2>
<p><a href="https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/jhu-csse.html"><span class="primary"><strong>JHU CSSE COVID cases per 100k </strong></span></a> estimates the daily number of new confirmed COVID-19 cases per 100,000 population, averaged over the past 7 days.</p>

<img data-src="day1-morning_files/figure-revealjs/examples-panel-covid2-1.svg" class="quarto-figure quarto-figure-center r-stretch"><aside class="notes">
<ul>
<li>WA switch to weekly reporting in 2022</li>
<li>FL reports ‚Äúwhenever‚Äù (weekly, biweekly, three days in a row, then 4 zeros, etc.)</li>
<li>API calculates change from cumulative, so no-report becomes a 0.</li>
<li>If state decreases total, then we see a negative.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<!--

## Examples of panel data - CHNG-CLI

[[**Change Healthcare COVID-like illness (CHNG-CLI)**]{.primary}](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/chng.html) reports the percentage of outpatient visits for COVID-related symptoms, based on deidentified Change Healthcare claims data.

::: {.cell layout-align="center"}
::: {.cell-output-display}
![](day1-morning_files/figure-revealjs/examples-chng-cli-1.svg){fig-align='center'}
:::
:::

## Examples of panel data - CHNG-COVID

[[**Change Healthcare COVID (CHNG-COVID)**]{.primary}](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/chng.html) reports the percentage of outpatient visits with confirmed COVID-19, based on Change Healthcare claims data.

::: {.cell layout-align="center"}
::: {.cell-output-display}
![](day1-morning_files/figure-revealjs/examples-chng-covid-1.svg){fig-align='center'}
:::
:::

-->
<!--  Numerator = denote the Covid counts. Denominator be the total count of visits. Scaling by population is not necessary here because the signal is already normalized to the total number of visits, which acts as a proxy for population size. -->
<!--

## Examples of panel data - HHS Admissions

[[**Confirmed COVID-19 Hospital Admissions per 100k**]{.primary}](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/hhs.html) estimates the daily sum of adult and pediatric confirmed COVID-19 hospital admissions, per 100,000 population, averaged over the past 7 days.

::: {.cell layout-align="center"}
::: {.cell-output-display}
![](day1-morning_files/figure-revealjs/examples-hhs-admissions-1.svg){fig-align='center'}
:::
:::
-->
<!--
## All together - Visualizing multiple panel data signals
Example: gathering different signals + visualizing panel data

::: {.cell layout-align="center"}

:::

## All together - Visualizing multiple panel data signals
Example: gathering different signals + **scaling** + visualizing panel data

::: {.cell layout-align="center"}

:::


::: {.cell layout-align="center"}

:::
<div style="text-align: center;">
<small>[Figure 1 from Reinhart et al. (2021)](https://www.pnas.org/doi/10.1073/pnas.2111452118)</small>
</div>

**Takeaway:** The auxiliary signals track changes in the official reported cases quite well. This is clearer when they have all been placed on the same range as reported cases per 100,000 people.
-->
<!-- Not panel data, 2 signals, same location, over time (bivariate time-series)
## Examples of panel data - COVID-19 cases and deaths in CA
* [**Takeaway**]{.primary}: Cases appear to strongly correlate with deaths several weeks later.
* We'll see this again in an upcoming session...

::: {.cell layout-align="center"}
::: {.cell-output-display}
![](day1-morning_files/figure-revealjs/plot-ca-cases-deaths-1.svg){fig-align='center'}
:::
:::

-->
</section></section>
<section>
<section id="versioned-data" class="title-slide slide level1 center" data-number="3">
<h1><span class="header-section-number">3</span> Versioned Data</h1>

</section>
<section id="intro-to-versioned-data" class="slide level2">
<h2>Intro to versioned data</h2>
<ul>
<li><p>Many epidemic aggregates are subject to <span class="primary">reporting delays and revisions</span></p></li>
<li><p>This is because individual-level data has delayed availability:</p></li>
</ul>
<p><span class="fourth-colour">Person comes to ER ‚Üí Admitted ‚Üí Has some tests ‚Üí Tests come back ‚Üí Entered into the system ‚Üí ‚Ä¶</span></p>
<ul>
<li><p>So, a ‚ÄúHospital admission‚Äù may not attributable to a particular condition until a few days have passed (the patient may even have been released)</p></li>
<li><p>Aggregated data have a longer pipeline from the incident to the report.</p></li>
<li><p>So we have to track both: <span class="primary">when the event occurred</span> and <span class="primary">when it was reported</span></p></li>
<li><p>Additionally, various mistakes lead to revisions</p></li>
<li><p>This means there can be <span class="primary">many different values</span> for the same date</p></li>
</ul>
</section>
<section id="versioned-data-1" class="slide level2">
<h2>Versioned data</h2>
<ul>
<li><p>The event time is indicated by <code>time_value</code> (aka <code>reference_date</code>)</p></li>
<li><p>Now, we add a second time index to indicate the data <code>version</code> (aka <code>reporting_date</code>)</p></li>
<li><p><code>version</code> = the time at which we saw a particular <code>value</code> associated to a <code>time_value</code> ::: {.cell layout-align=‚Äúcenter‚Äù} ::: {.cell-output .cell-output-stdout}</p></li>
</ul>
<pre><code># A tibble: 6 √ó 4
  time_value geo_value percent_cli version   
  &lt;date&gt;     &lt;chr&gt;           &lt;dbl&gt; &lt;date&gt;    
1 2020-06-01 ca               2.14 2020-06-06
2 2020-06-01 ca               2.14 2020-06-08
3 2020-06-01 ca               2.11 2020-06-09
4 2020-06-01 ca               2.13 2020-06-10
5 2020-06-01 ca               2.20 2020-06-11
6 2020-06-01 ca               2.23 2020-06-12</code></pre>
<p>::: :::</p>
<ul>
<li>Note that this feature can be indicated in different ways (ex. <code>version</code>, <code>issue</code>, <code>release</code>, <code>as_of</code>).</li>
</ul>
</section>
<section id="versioned-panel-data" class="slide level2">
<h2>Versioned panel data</h2>
<p>Estimated percentage of outpatient visits due to CLI across multiple versions.</p>
<!--
with updates and revisions to past data as new issue dates are released:
-->

<img data-src="day1-morning_files/figure-revealjs/versioned-panel-multi-states-ex-2-1.svg" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="latency-and-revision-in-signals" class="slide level2">
<h2>Latency and revision in signals</h2>
<ul>
<li><span class="primary"><strong>Revised</strong></span> data is updated or corrected after initial publication</li>
</ul>
<p><strong>Example</strong>: COVID-19 case reports are revised reporting backlogs are cleared</p>
<ul>
<li><span class="primary"><strong>Latency</strong></span> the delay between data collection and availability</li>
</ul>
<p><strong>Example</strong>: A signal based on insurance claims may take several days to appear as claims are processed</p>
<!--
## Revision triangle, Outpatient visits in WA 2022

* 7-day trailing average to smooth day-of-week effects

::: {.cell layout-align="center"}
::: {.cell-output-display}
![](day1-morning_files/figure-revealjs/revision-triangle-1.png){fig-align='center'}
:::
:::
-->
</section>
<section id="revisions" class="slide level2">
<h2>Revisions</h2>
<p>Many data sources are subject to revisions:</p>
<ul>
<li><p>Case and death counts are frequently corrected or adjusted by authorities</p></li>
<li><p>Medical claims can take weeks to be submitted and processed</p></li>
</ul>
<!--* In the previous slide's example because doctor's visits may be reported to the health system partners several days after they occur, these signals are typically available with several days of lag. This means that estimates for a specific day are only available several days later. -->
<ul>
<li><p>Lab tests and medical records can be backlogged</p></li>
<li><p>Surveys are not completed promptly</p></li>
</ul>
<p><span class="primary">An accurate revision log is crucial for researchers building forecasts</span></p>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Obvious but crucial</strong></p>
</div>
<div class="callout-content">
<p>A forecast that is made today can only use data we have access to today</p>
</div>
</div>
</div>
</section>
<section id="three-types-of-revisions" class="slide level2">
<h2>Three types of revisions</h2>
<ol type="1">
<li><span class="primary"><strong>Sources that don‚Äôt revise</strong></span> (provisional and final are the same)</li>
</ol>
<p>Google symptoms</p>
<ol start="2" type="1">
<li><span class="secondary"><strong>Predictable revisions</strong></span></li>
</ol>
<p>Claims data (CHNG) and public health reports aligned by test, hospitalization, or death date</p>
<p>Almost always revised upward as additional claims enter the pipeline</p>
<ol start="3" type="1">
<li><span class="tertiary"><strong>Revisions that are large and erratic to predict</strong></span></li>
</ol>
<p>COVID cases and deaths</p>
<p>These are aligned by report date</p>
<!-- (which can be highly variable & less predictable compared to test data). -->
</section>
<section id="types-of-revisions---comparison-between-2.-and-3." class="slide level2">
<h2>Types of revisions - Comparison between 2. and 3.</h2>
<ul>
<li>Revision behavior for two indicators in the HRR containing Charlotte, NC.</li>
</ul>
<!-- Each colored line corresponds to the data as reported on a particular date (as of dates varying from 28 September through 19 October). -->
<ul>
<li><p><span class="secondary"><strong>DV-CLI signal (left)</strong></span>: regularly revised, but effects fade</p></li>
<li><p><span class="tertiary"><strong>JHU CSSE cases (right)</strong></span> remain ‚Äúas first reported‚Äù until a major correction is made on Oct.&nbsp;19</p></li>
</ul>

<!--
<div style="text-align: center;">
<small>[Figure 1 from McDonald et al. (2021)](https://www.pnas.org/doi/pdf/10.1073/pnas.2111453118)</small>
</div>
-->
<img data-src="day1-morning_files/figure-revealjs/fig1-McDonald-1.svg" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="key-takeaways" class="slide level2">
<h2>Key takeaways</h2>
<dl>
<dt>Medical claims revisions</dt>
<dd>
More systematic and predictable
</dd>
</dl>
<p><br><br></p>
<dl>
<dt>COVID-19 case report revisions</dt>
<dd>
Erratic and often unpredictable
</dd>
</dl>
<p><br><br></p>
<dl>
<dt>Large spikes or anomalies can occur as</dt>
<dd>
Reporting backlogs are cleared
</dd>
<dd>
Changes in case definitions are implemented
</dd>
</dl>
<!--
## Reporting backlogs - Example

In Bexar County, Texas, during the summer of 2020...

* Large backlog of case reports results in a spike
* Auxilliary signals show no such dramatic increase
* Reports themselves may not be trustworthy without context


* **Left**: On July 16, 4,810 [backlogged cases]{.primary} were reported, reflecting a 2-week delay. This caused a prolonged spike due to the 7-day trailing average applied to the counts.

* **Right**: CTIS estimates of CLI-in-community showed more stable underlying trends.


::: {.cell layout-align="center"}
::: {.cell-output-display}
![](day1-morning_files/figure-revealjs/fig4-Reinhart-1.svg){fig-align='center'}
:::
:::
-->
<!--
## Reporting backlogs - Key takeaways

<br><br>

* [Reporting issues common across U.S. jurisdictions]{.primary}

<br><br>

* [Audits regularly discovered misclassified or unreported cases and deaths]{.primary}

<br><br>

* [Cross-checking data with external sources from different reporting systems]{.primary}
-->
</section></section>
<section>
<section id="epidata-repository-and-api" class="title-slide slide level1 center" data-number="4">
<h1><span class="header-section-number">4</span> Epidata Repository and API</h1>

</section>
<section id="what-is-the-epidata-repository" class="slide level2">
<h2>What is the Epidata repository</h2>
<p><span class="primary"><strong>Epidata:</strong></span> repository of aggregated epi-surveillance time series</p>
<p>Code is open-source. Most indicators are public.</p>
<ul>
<li><p>To date, it has accumulated over 5 billion records.</p></li>
<li><p>At the peak of the pandemic, handled millions of API queries per day.</p></li>
<li><p>Many aren‚Äôt available elsewhere</p></li>
</ul>
<dl>
<dt>Data from</dt>
<dd>
public health reporting, medical insurance claims, medical device data, Google search queries, wastewater, app-based mobility patterns.
</dd>
</dl>
<p><br></p>
<dl>
<dt>Added value</dt>
<dd>
revision tracking, anomaly detection, trend detection, smoothing, imputation, geo-temporal-demographic disaggregation.
</dd>
</dl>
<!--
## Goals of Delphi Epidata platform and repository

1. **Provide many aggregated epi-surveillance time-series ("epi-signals")**
    + Mirror signals from other sources, especially if revisions are not tracked
    + Be the national historical repository of record & preserve the raw data

Hence: include also signals available elsewhere, especially if they don't keep data revisions - E.g. CDC's own NSSP, NWSS

2. **Be the go-to place for epi-signal discovery, including those held elsewhere**

3. **Add value to existing signals and synthesize new ones**
    + Via signal fusion, nowcasting, smoothing


::: {.callout-important appearance="simple"}
Make epi-surveillance more nimble, complete, standardized, robust, and real-time
:::

-->
</section>
</section>
<section>
<section id="epidatr" class="title-slide slide level1 center" data-number="5">
<h1><span class="header-section-number">5</span> <code>{epidatr}</code></h1>

</section>
<section id="installing-epidatr" class="slide level2">
<h2>Installing <code>{epidatr}</code></h2>
<p>(you already did this, but just for posterity‚Ä¶)</p>
<p>Install the CRAN version</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a></a><span class="co"># Install the CRAN version</span></span>
<span id="cb6-2"><a></a>pak<span class="sc">::</span><span class="fu">pkg_install</span>(<span class="st">"epidatr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br></p>
<p>or the development version</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a></a><span class="co"># Install the development version from the GitHub dev branch</span></span>
<span id="cb7-2"><a></a>pak<span class="sc">::</span><span class="fu">pkg_install</span>(<span class="st">"cmu-delphi/epidatr@dev"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The CRAN listing is <a href="https://cran.r-project.org/package=epidatr/index.html">here</a>.</p>
<!--
## Python

In Python, install [`delphi-epidata` from PyPI](https://pypi.org/project/delphi-epidata/) with

``` sh
pip install delphi-epidata
```

<br>

`delphi-epidata` is soon to be replaced with `epidatpy`.

``` sh
# Latest dev version
pip install -e "git+https://github.com/cmu-delphi/epidatpy.git#egg=epidatpy"

# PyPI version (not yet available)
pip install epidatpy
```
-->
</section>
<section id="api-keys" class="slide level2">
<h2>API keys</h2>
<ul>
<li><p>Anyone may access the Epidata API anonymously without providing any personal data!!</p></li>
<li><p>Anonymous API access is subject to some restrictions: <small>public datasets only; 60 requests per hour; only two parameters may have multiple selections</small></p></li>
<li><p>API key grants privileged access; can be obtained by <a href="https://api.delphi.cmu.edu/epidata/admin/registration_form">registering with us</a></p></li>
<li><p>Privileges of registration: no rate limit; no limit on multiple selections</p></li>
<li><p>We just want to know which signals people care about and ensure we‚Äôre providing benefit</p></li>
</ul>
<!-- rate limited to 60 requests per hour;  -->
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Tip</strong></p>
</div>
<div class="callout-content">
<ul>
<li>The <code>{epidatr}</code> client automatically searches for the key in the <code>DELPHI_EPIDATA_KEY</code> environment variable.</li>
<li>We recommend storing it in your <code>.Renviron</code> file, which R reads by default.</li>
<li>More on setting your API key <a href="https://rdrr.io/cran/epidatr/man/get_api_key.html">here</a>.</li>
</ul>
</div>
</div>
</div>
</section>
<section id="using-epidatr" class="slide level2">
<h2>Using <code>{epidatr}</code></h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a></a><span class="fu">library</span>(epidatr)</span>
<span id="cb8-2"><a></a>hhs_flu_nc <span class="ot">&lt;-</span> <span class="fu">pub_covidcast</span>(</span>
<span id="cb8-3"><a></a>  <span class="at">source =</span> <span class="st">'hhs'</span>,</span>
<span id="cb8-4"><a></a>  <span class="at">signals =</span> <span class="st">'confirmed_admissions_influenza_1d'</span>,</span>
<span id="cb8-5"><a></a>  <span class="at">geo_type =</span> <span class="st">'state'</span>,</span>
<span id="cb8-6"><a></a>  <span class="at">time_type =</span> <span class="st">'day'</span>,</span>
<span id="cb8-7"><a></a>  <span class="at">geo_values =</span> <span class="st">'nc'</span>,</span>
<span id="cb8-8"><a></a>  <span class="at">time_values =</span> <span class="fu">c</span>(<span class="dv">20240401</span>, <span class="dv">20240405</span><span class="sc">:</span><span class="dv">20240414</span>)</span>
<span id="cb8-9"><a></a>)</span>
<span id="cb8-10"><a></a><span class="fu">head</span>(hhs_flu_nc, <span class="at">n =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 √ó 15
  geo_value signal     source geo_type time_type time_value direction issue     
  &lt;chr&gt;     &lt;chr&gt;      &lt;chr&gt;  &lt;fct&gt;    &lt;fct&gt;     &lt;date&gt;         &lt;dbl&gt; &lt;date&gt;    
1 nc        confirmed‚Ä¶ hhs    state    day       2024-04-01        NA 2024-04-22
2 nc        confirmed‚Ä¶ hhs    state    day       2024-04-05        NA 2024-04-22
3 nc        confirmed‚Ä¶ hhs    state    day       2024-04-06        NA 2024-04-22
# ‚Ñπ 7 more variables: lag &lt;dbl&gt;, missing_value &lt;dbl&gt;, missing_stderr &lt;dbl&gt;,
#   missing_sample_size &lt;dbl&gt;, value &lt;dbl&gt;, stderr &lt;dbl&gt;, sample_size &lt;dbl&gt;</code></pre>
</div>
</div>
<p><br></p>
<div class="callout callout-important no-icon callout-style-simple">
<div class="callout-body">
<div class="callout-content">
<p>The <code>covidcast</code> endpoint supports <code>*</code> in its time and geo fields.</p>
<p>Signal values for all available counties: replace <code>geo_values = "06059"</code> with <code>geo_values = "*"</code>.</p>
</div>
</div>
</div>
</section>


<section id="returned-data---covidcast-main-endpoint" class="slide level2">
<h2>Returned data - COVIDcast main endpoint</h2>
<p><code>pub_covidcast()</code> outputs a tibble, where each row represents one observation</p>
<p>Each observation is aggregated by time and by geographic region</p>
<ol type="1">
<li><code>time_value</code>: time period when the events occurred.</li>
<li><code>geo_value</code>: geographic region where the events occurred.</li>
<li><code>value</code>: estimated value.</li>
<li><code>stderr</code>: standard error of the estimate, usually referring to the sampling error.</li>
<li><code>sample_size</code>: number of events used in the estimation.</li>
</ol>
<!--For example, a number of COVID-19 antigen tests were performed in the state of New York on August 1. The `time_value` would be August 1, with `geo_value` indicating the state of New York, while the remaining fields would give the estimated test positivity rate (the percentage of tests that were positive for COVID-19), its standard error, and the number of tests used to calculate the estimate.-->
</section>
<section id="returned-data---covidcast-main-endpoint-1" class="slide level2">
<h2>Returned data - COVIDcast main endpoint</h2>
<p><span class="primary">Also reports</span></p>
<ul>
<li><p><code>issue</code>: The date this observation was published</p></li>
<li><p><code>lag</code>: The period between when the events occurred and when the observation was published</p></li>
</ul>
<p>Tracks the complete revision history of the signal</p>
<p>Allows for historical reconstructions of information that was available at a specific times</p>
<p><span class="primary"><strong>More on this soon!</strong></span></p>
<!-- * Meaning that unlike most other sources of COVID data, it tracks the complete revision history of the signal. -->
</section>

</section>
<section>
<section id="versioning-in-epidatr" class="title-slide slide level1 center" data-number="6">
<h1><span class="header-section-number">6</span> Versioning in <code>{epidatr}</code></h1>

</section>
<section id="versioned-data-in-epidatr" class="slide level2">
<h2>Versioned data in <code>{epidatr}</code></h2>
<p>Epidata API contains each signal‚Äôs estimate, location, date, and update timeline</p>
<p>Requesting Specific Data Versions:</p>
<ul>
<li>Use <code>as_of</code> or <code>issues</code> to specify data availability</li>
<li><code>as_of</code> always fetches one version</li>
<li><code>issues</code> can fetch multiple</li>
<li>Only one may be used at a time</li>
</ul>
</section>
<section id="obtaining-data-as-of-a-specific-date" class="slide level2">
<h2>Obtaining data ‚Äúas of‚Äù a specific date</h2>
<p><a href="https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/doctor-visits.html">Doctor Visits</a> (from the <code>covidcast</code> endpoint)</p>
<ul>
<li>The percentage of outpatient visits w/ Covid-like illness</li>
<li>Pennsylvania on May 1, 2020:</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a></a>dv_pa_as_of <span class="ot">&lt;-</span> <span class="fu">pub_covidcast</span>(</span>
<span id="cb16-2"><a></a>  <span class="at">source =</span> <span class="st">"doctor-visits"</span>,</span>
<span id="cb16-3"><a></a>  <span class="at">signals =</span> <span class="st">"smoothed_adj_cli"</span>,</span>
<span id="cb16-4"><a></a>  <span class="at">time_type =</span> <span class="st">"day"</span>,</span>
<span id="cb16-5"><a></a>  <span class="at">time_values =</span> <span class="st">"2020-05-01"</span>,</span>
<span id="cb16-6"><a></a>  <span class="at">geo_type =</span> <span class="st">"state"</span>,</span>
<span id="cb16-7"><a></a>  <span class="at">geo_values =</span> <span class="st">"pa"</span>,</span>
<span id="cb16-8"><a></a>  <span class="at">as_of =</span> <span class="st">"2020-05-07"</span></span>
<span id="cb16-9"><a></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 √ó 7
  geo_value signal           source        time_value issue        lag value
  &lt;chr&gt;     &lt;chr&gt;            &lt;chr&gt;         &lt;date&gt;     &lt;date&gt;     &lt;dbl&gt; &lt;dbl&gt;
1 pa        smoothed_adj_cli doctor-visits 2020-05-01 2020-05-07     6  2.58</code></pre>
</div>
</div>
<ul>
<li>Initial estimate <em>issued</em> on May 7, 2020</li>
<li>Due to delay from reporting and ingestion by the API</li>
</ul>
</section>

<section id="versioning-is-important-for-forecasting" class="slide level2">
<h2>Versioning is important for forecasting</h2>
<p><br></p>
<ul>
<li>Backtesting requires using data that would have been available at the time</li>
</ul>
<p><br></p>
<ul>
<li>Not later updates</li>
</ul>
<p><br></p>
<ul>
<li>Overly optimistic</li>
</ul>
<!--
## Obtaining multiple specific issues for one state
Request all issues in a certain time period

::: {.cell layout-align="center"}

```{.r .cell-code  code-line-numbers="1,8,9"}
dv_pa_issues <- pub_covidcast(
  source = "doctor-visits",
  signals = "smoothed_adj_cli",
  time_type = "day",
  time_values = "2020-05-01",
  geo_type = "state",
  geo_values = "pa",
  issues = epirange("2020-05-01", "2020-05-15")
)
```
:::

::: {.cell layout-align="center"}
::: {.cell-output .cell-output-stdout}

```
# A tibble: 6 √ó 8
  geo_value signal           source     time_value issue        lag value stderr
  <chr>     <chr>            <chr>      <date>     <date>     <dbl> <dbl>  <dbl>
1 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-07     6  2.58     NA
2 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-08     7  3.28     NA
3 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-09     8  3.32     NA
4 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-12    11  3.59     NA
5 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-13    12  3.63     NA
6 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-14    13  3.66     NA
```


:::
:::

## Obtaining multiple issues for one state

To get all issues up to a specific date, set an extreme lower bound

::: {.cell layout-align="center"}

```{.r .cell-code  code-line-numbers="1,8,9"}
dv_pa_issues_sub <- pub_covidcast(
  source = "doctor-visits",
  signals = "smoothed_adj_cli",
  time_type = "day",
  time_values = "2020-05-01",
  geo_type = "state",
  geo_values = "pa",
  issues = epirange("1900-01-01", "2020-05-15")
)
```
:::

::: {.cell layout-align="center"}
::: {.cell-output .cell-output-stdout}

```
# A tibble: 6 √ó 8
  geo_value signal           source     time_value issue        lag value stderr
  <chr>     <chr>            <chr>      <date>     <date>     <dbl> <dbl>  <dbl>
1 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-07     6  2.58     NA
2 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-08     7  3.28     NA
3 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-09     8  3.32     NA
4 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-12    11  3.59     NA
5 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-13    12  3.63     NA
6 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-14    13  3.66     NA
```


:::
:::

No change here ‚Ä¢ Can matter if the latency or reporting lag is unknown

[API docs](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/doctor-visits.html)
show the earliest date available.

## Obtaining multiple issues for one state

At some point, nothing changes ‚Ä¢ It is [finalized]{.primary} ‚Ä¢ That will be the "last" issue

::: {.cell layout-align="center"}

```{.r .cell-code  code-line-numbers="1,8,9"}
dv_pa_issues_all <- pub_covidcast(
  source = "doctor-visits",
  signals = "smoothed_adj_cli",
  time_type = "day",
  time_values = "2020-05-01",
  geo_type = "state",
  geo_values = "pa",
  issues = epirange("1900-01-01", "2024-12-11") # From the 1900s to today
)
```
:::

::: {.cell layout-align="center"}
::: {.cell-output .cell-output-stdout}

```
# A tibble: 6 √ó 8
  geo_value signal           source     time_value issue        lag value stderr
  <chr>     <chr>            <chr>      <date>     <date>     <dbl> <dbl>  <dbl>
1 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-06-29    59  5.99     NA
2 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-06-30    60  5.99     NA
3 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-01    61  5.95     NA
4 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-02    62  5.97     NA
5 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-03    63  5.97     NA
6 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-04    64  5.97     NA
```


:::
:::

* Avoid queries with too-late minimum too-early maximum issue
* Could be misleading results

* We caution against starting queries with too late a minimum issue or too early maximum issue, as it could lead to incorrect or misleading results. You're safest bet to capture all issues is on the next slide...

## Obtaining all issues for one state


::: {.cell layout-align="center"}

```{.r .cell-code  code-line-numbers="1,5,8,9"}
dv_pa_issues_star <- pub_covidcast(
  source = "doctor-visits",
  signals = "smoothed_adj_cli",
  time_type = "day",
  time_values = epirange("2020-05-01", "2020-05-07"),
  geo_type = "state",
  geo_values = "pa",
  issues = "*"
)
```
:::

::: {.cell layout-align="center"}
::: {.cell-output .cell-output-stdout}

```
# A tibble: 8 √ó 8
  geo_value signal           source     time_value issue        lag value stderr
  <chr>     <chr>            <chr>      <date>     <date>     <dbl> <dbl>  <dbl>
1 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-07     6  2.58     NA
2 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-08     7  3.28     NA
3 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-09     8  3.32     NA
4 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-12    11  3.59     NA
5 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-13    12  3.63     NA
6 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-14    13  3.66     NA
7 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-15    14  3.66     NA
8 pa        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-16    15  3.61     NA
```


:::
:::

## Obtaining all issues for all states

Using `*` gives all available

::: {.cell layout-align="center"}

```{.r .cell-code  code-line-numbers="1,7-9"}
dv_state_issues_star <- pub_covidcast(
  source = "doctor-visits",
  signals = "smoothed_adj_cli",
  time_type = "day",
  time_values = epirange("2020-05-01", "2020-05-07"),
  geo_type = "state",
  geo_values = "*",
  issues = "*"
)
```
:::

::: {.cell layout-align="center"}
::: {.cell-output .cell-output-stdout}

```
# A tibble: 6 √ó 8
  geo_value signal           source     time_value issue        lag value stderr
  <chr>     <chr>            <chr>      <date>     <date>     <dbl> <dbl>  <dbl>
1 ak        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-07     6  1.61     NA
2 ak        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-08     7  2.40     NA
3 ak        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-09     8  2.38     NA
4 ak        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-12    11  2.38     NA
5 ak        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-13    12  2.36     NA
6 ak        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-05-14    13  2.36     NA
```


:::
:::


<!--

## Last but not least... The do nothing approach

![](gfx/do-nothing.jpg){height="550px"}



## The do nothing approach

* `geo_values = *` is the default
* `issues` defaults to "most recent"


::: {.cell layout-align="center"}

```{.r .cell-code}
dv_state_default <- pub_covidcast(
  source = "doctor-visits",
  signals = "smoothed_adj_cli",
  time_type = "day",
  time_values = epirange("2020-05-01", "2020-05-07"),
  geo_type = "state"
)
```
:::

-->
<!--
## Obtaining one issue for all states
-->
<!--
[**Final question:**]{.primary}  What do you think happens when we adopt a "do nothing" approach to `geo_values` and `issues` (take both of them out)?

<small> [**Hint**]{.primary}: remember a couple slides ago, when we removed `as_of`, we got the most recent estimate for PA. </small>
-->
<!--
Defaults are intended to be "what you would expect"

::: {.cell layout-align="center"}

```{.r .cell-code}
dv_state_default <- pub_covidcast(
  source = "doctor-visits",
  signals = "smoothed_adj_cli",
  time_type = "day",
  time_values = epirange("2020-05-01", "2020-05-07"),
  geo_type = "state"
)
```
:::

::: {.cell layout-align="center"}
::: {.cell-output .cell-output-stdout}

```
# A tibble: 6 √ó 8
  geo_value signal           source     time_value issue        lag value stderr
  <chr>     <chr>            <chr>      <date>     <date>     <dbl> <dbl>  <dbl>
1 ak        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-04    64  5.72     NA
2 al        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-04    64  2.74     NA
3 ar        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-04    64  4.23     NA
4 az        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-04    64  2.78     NA
5 ca        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-04    64  4.25     NA
6 co        smoothed_adj_cli doctor-vi‚Ä¶ 2020-05-01 2020-07-04    64  8.77     NA
```


:::
:::

<hr>

* most recent issue
* all states


<!-- ## Observations issued with a specific lag
<div style="font-size: 0.8em;">
* We can use the `lag` argument to request only data reported with a certain lag.

* **Example**: Request  a lag of 7 days fetches only data issued exactly 7 days after the corresponding `time_value`:

::: {.cell layout-align="center"}

:::

::: {.cell layout-align="center"}

:::
</div>

## Query results exclusion
<div style="font-size: 0.8em;">
* Although the query we ran on the previous slide requested values from May 1 to May 7, May 3 and May 4 were not included due to a 7-day lag.

* Results for those dates appear only if updates are issued on the corresponding lag day (e.g., May 10).

::: {.cell layout-align="center"}

:::

::: {.cell layout-align="center"}

:::
</div>
-->
</section>
</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<p><img src="gfx/delphi.jpg" class="slide-logo"></p>
<div class="footer footer-default">
<p>Understanding Data ‚Äî <a href="https://github.com/cmu-delphi/micom-tooling-workshop-2025">cmu-delphi/micom-tooling-workshop-2025</a></p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: false,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1600,

        height: 900,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/cmu-delphi\.github\.io\/micom-tooling-workshop-2025\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
                // target, if specified
                link.setAttribute("target", "_blank");
                if (link.getAttribute("rel") === null) {
                  link.setAttribute("rel", "noopener");
                }
                // default icon
                link.classList.add("external");
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>